32_awk_lab/awklab.txt

```diff
@@ -1,11 +1,12 @@
 1. Write an awk one-liner that prints the file "shells" with a blank line after each line
-Answer: 
+Answer: cat /etc/shells | awk ' // { print $0, "\n" } '
 
 
 2. Write an awk one-liner that prints the file "shells" with a ">" sign at the beginning of each line
-Answer: 
+Answer: cat /etc/shells | awk ' // { print "> ", $0 } '
 
 3. Write an awk one-liner to swap the first and last words in the output of "help cd"
+Answer: help cd | awk ' { printf("%s", $NF) } { for(i=2; i<NF; i++) printf(" %s", $i) } { if(NF > 1) printf(" %s", $1) } { printf("\n") } '
```

Comment:
Note that```$NF``` will give the value of the last column (technically it will give you the value in column number ```NF```), and ```$1``` will give you the value of the first column. 

So we can just swap them:
```help cd | awk '{temp=$1; $1=$NF; $NF=temp}; {print}'```